conan/api/subapi/remotes.py

```diff
@@ -48,42 +48,39 @@ def disable(self, pattern):
         remotes = self.list(pattern, only_enabled=False)
         for r in remotes:
             r.disabled = True
-            self.update(r)
+            self.update(r.name, disabled=True)
         return remotes
 
     def enable(self, pattern):
         remotes = self.list(pattern, only_enabled=False)
         for r in remotes:
             r.disabled = False
-            self.update(r)
+            self.update(r.name, disabled=False)
         return remotes
 
     def get(self, remote_name):
         app = ConanApp(self.conan_api.cache_folder)
         return app.cache.remotes_registry.read(remote_name)
 
-    def add(self, remote: Remote, force=False):
+    def add(self, remote: Remote, force=False, index=None):
         app = ConanApp(self.conan_api.cache_folder)
-        app.cache.remotes_registry.add(remote, force=force)
+        app.cache.remotes_registry.add(remote, force=force, index=index)
 
-    def remove(self, remote_name):
+    def remove(self, pattern: str):
         app = ConanApp(self.conan_api.cache_folder)
-        remotes = self.list(remote_name, only_enabled=False)
+        remotes = self.list(pattern, only_enabled=False)
         for remote in remotes:
             app.cache.remotes_registry.remove(remote.name)
             users_clean(app.cache.localdb, remote.url)
 
-    def update(self, remote: Remote):
-        app = ConanApp(self.conan_api.cache_folder)
-        app.cache.remotes_registry.update(remote)
-
-    def move(self, remote: Remote, index: int):
+    def update(self, remote_name, url=None, secure=None, disabled=None, index=None):
```

Comment:
This API was changed, to not require passing a Remote object, but just the updates to do. The problem with passing the Remote object is that the update semantics are more implicit/hidden, and there can be more inconsistencies like coming with a new ``remote.name`` not existing in the current ones.